api/global/handler.go

```diff
@@ -0,0 +1,81 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package global
+
+import (
+	"log"
+	"os"
+	"sync"
+
+	"go.opentelemetry.io/otel/api/oterror"
+)
+
+var (
+	defaultHandler = &handler{
+		l: log.New(os.Stderr, "", log.LstdFlags),
+	}
+
+	// Ensure the handler implements oterror.Handle at build time.
+	_ oterror.Handler = (*handler)(nil)
+)
+
+// handler logs all errors to STDERR.
+type handler struct {
+	sync.Mutex
+	delegate oterror.Handler
+
+	l *log.Logger
+}
+
+func (h *handler) setDelegate(d oterror.Handler) {
+	h.Lock()
+	defer h.Unlock()
+	if h.delegate != nil {
+		// delegate already registered
+		return
+	}
+
+	h.delegate = d
+}
+
+// Handle implements oterror.Handler.
+func (h *handler) Handle(err error) {
+	if h.delegate != nil {
```

Comment:
I suspect the race detector would complain if you don't also read `h.delegate` under the lock. How about:
```
h.Lock()
d := h.delegate
h.Unlock()

if d != nil { ... }
h.l.Print(err)
```

Reply:
Yeah, agreed. I switched to just having the delegate be an `atomic.Value` instead. LMK what you think.